pkg/scheduler/framework/v1alpha1/framework_test.go

```diff
@@ -876,6 +876,454 @@ func TestFilterPlugins(t *testing.T) {
 	}
 }
 
+func TestPreBindPlugins(t *testing.T) {
+	tests := []struct {
+		name       string
+		plugins    []*TestPlugin
+		wantStatus *Status
+	}{
+		{
+			name:       "NoPreBindPlugin",
+			plugins:    []*TestPlugin{},
+			wantStatus: nil,
+		},
+		{
+			name: "SuccessPreBindPlugins",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin",
+					inj:  injectedResult{PreBindStatus: int(Success)},
+				},
+			},
+			wantStatus: nil,
+		},
+		{
+			name: "UnshedulablePreBindPlugin",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin",
+					inj:  injectedResult{PreBindStatus: int(Unschedulable)},
+				},
+			},
+			wantStatus: NewStatus(Error, `error while running "TestPlugin" prebind plugin for pod "": injected status`),
+		},
+		{
+			name: "ErrorPreBindPlugin",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin",
+					inj:  injectedResult{PreBindStatus: int(Error)},
+				},
+			},
+			wantStatus: NewStatus(Error, `error while running "TestPlugin" prebind plugin for pod "": injected status`),
+		},
+		{
+			name: "UnschedulablePreBindPlugin",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin",
+					inj:  injectedResult{PreBindStatus: int(UnschedulableAndUnresolvable)},
+				},
+			},
+			wantStatus: NewStatus(Error, `error while running "TestPlugin" prebind plugin for pod "": injected status`),
+		},
+		{
+			name: "SuccessErrorPreBindPlugins",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin",
+					inj:  injectedResult{PreBindStatus: int(Success)},
+				},
+				{
+					name: "TestPlugin 1",
+					inj:  injectedResult{PreBindStatus: int(Error)},
+				},
+			},
+			wantStatus: NewStatus(Error, `error while running "TestPlugin 1" prebind plugin for pod "": injected status`),
+		},
+		{
+			name: "ErrorSuccessPreBindPlugin",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin",
+					inj:  injectedResult{PreBindStatus: int(Error)},
+				},
+				{
+					name: "TestPlugin 1",
+					inj:  injectedResult{PreBindStatus: int(Success)},
+				},
+			},
+			wantStatus: NewStatus(Error, `error while running "TestPlugin" prebind plugin for pod "": injected status`),
+		},
+		{
+			name: "SuccessSuccessPreBindPlugin",
+			plugins: []*TestPlugin{
+				{
+					name: "TestPlugin",
+					inj:  injectedResult{PreBindStatus: int(Success)},
+				},
+				{
+					name: "TestPlugin 1",
+					inj:  injectedResult{PreBindStatus: int(Success)},
+				},
+			},
+			wantStatus: nil,
+		},
+		{
+			name: "ErrorErrorPreBindPlugin",
```

Comment:
This test name looks weird, do we have a better name to describe this test?

Reply:
ErrorAndErrorPlugins - is this okay?